cpp/include/raft/util/inline.hpp

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#ifdef RAFT_COMPILED
+#define RAFT_INLINE_CONDITIONAL
```

Comment:
Can you please use the macros in `include/core/detail/macros.hpp` for this? This isn't really a utiliity.

Reply:
Yes. I have moved the definition of `RAFT_INLINE_CONDITIONAL` to `macros.hpp` and removed `inline.hpp`.